src/net/mod.rs

```diff
@@ -57,6 +57,17 @@ impl RecordHeaders {
         req.header(OFFSET_HEADER_NAME.clone(), self.offset)
             .header(DATA_SIZE_HEADER_NAME.clone(), self.data_size)
     }
+
+    /// # Errors
+    /// Returns an error if the body's length does not align with the data-size header.
+    /// Namely, checks if `body.len() % data-size == 0`
+    pub fn matches_body(&self, body_len: usize) -> Result<(), MpcServerError> {
+        (body_len % self.data_size as usize == 0)
```

Comment:
please please never trust your inputs to be within the valid range. This will crash the server if `data_size` is 0. We could argue whether we should return $\bot$ or detailed explanation (I strongly advise to be as detailed as possible in debug builds and tell the caller what exactly went wrong) in this case, but everything you receive must be sanitized properly.

Reply:
of course...adding an extra check on `data_size`